docs/design-documents/platform/pin-names/pin_names.md

```diff
@@ -0,0 +1,198 @@
+
+    // SPI signals are aliased as below  
+    
+    SPI_CS = D10,   // SPI CS  
+    SPI_MOSI = D11, // SPI MOSI  
+    SPI_MISO = D12, // SPI MISO  
+    SPI_SCK = D13,  // SPI SCK
+
+### Custom/Vendor Header Pins
+
+**Custom/Vendor Connector I2C Definition**
+
+All I2C definitions for a new connector should be defined as follows:
+
+    // ConnectorName is the name of the specific connector you are implementing signals for here  
+    // Px_xx relates to the processor pin connected to the custom/vendor connector pin  
+    
+    ConnectorName_I2C0_SCL  = Px_xx,  // Connector I2C0 SCL  
+    ConnectorName_I2C0_SDA  = Px_xx,  // Connector I2C0 SDA  
+    ConnectorName_I2C1_SCL  = Px_xx,  // Connector I2C1 SCL  
+    ConnectorName_I2C1_SDA  = Px_xx,  // Connector I2C1 SDA  
+    .  
+    .  
+    .  
+    ConnectorName_I2CN_SCL  = Px_xx,  // Connector I2CN SCL  
+    ConnectorName_I2CN_SDA  = Px_xx,  // Connector I2CN SDA  
+
+**Custom/Vendor Connector SPI Definition**
+
+All SPI definitions for a new connector should be defined as follows:
+
+    // ConnectorName is the name of the specific connector you are implementing signals for here  
+    // Px_xx relates to your processors pin connected to the custom/vendor connector pin
+    
+    ConnectorName_SPI0_CS   = Px_xx,  // Connector SPI0 CS  
+    ConnectorName_SPI0_MOSI = Px_xx,  // Connector SPI0 MOSI  
+    ConnectorName_SPI0_MISO = Px_xx,  // Connector SPI0 MISO  
+    ConnectorName_SPI0_SCK  = Px_xx,  // Connector SPI0 SCK  
+    
+    ConnectorName_SPI1_CS   = Px_xx,  // Connector SPI1 CS  
+    ConnectorName_SPI1_MOSI = Px_xx,  // Connector SPI1 MOSI  
+    ConnectorName_SPI1_MISO = Px_xx,  // Connector SPI1 MISO  
+    ConnectorName_SPI1_SCK  = Px_xx,  // Connector SPI1 SCK  
+    .  
+    .  
+    .  
+    
+    ConnectorName_SPIN_CS   = Px_xx,  // Connector SPIN CS  
+    ConnectorName_SPIN_MOSI = Px_xx,  // Connector SPIN MOSI  
+    ConnectorName_SPIN_MISO = Px_xx,  // Connector SPIN MISO  
+    ConnectorName_SPIN_SCK  = Px_xx,  // Connector SPIN SCK  
+
+### Board Components Pins
+
+**LED Definition**
+
+All LED definitions should be defined as follows:
+
+    // Px_xx relates to the processor pin connected to the LED
+    
+    LED1 = Px_xx,  // LED1  
+    LED2 = Px_xx,  // LED2  
+    LED3 = Px_xx,  // LED3  
+    LED4 = Px_xx,  // LED4  
+    .  
+    .  
+    LEDN = Px_xx   // LEDN
+
+**BUTTON Definition**
+
+All BUTTON definitions should be defined as follows:
+
+    // Px_xx relates to the processor pin connected to the Button  
+    BUTTON1 = Px_xx,  // BUTTON1  
+    BUTTON2 = Px_xx,  // BUTTON2  
+    BUTTON3 = Px_xx,  // BUTTON3  
+    BUTTON4 = Px_xx,  // BUTTON4   
+    .  
+    .  
+    BUTTONN = Px_xx   // BUTTONN  
+
+# Other information
+
+### Avoid Unnecessary Definitions
+
+If only 1 LED/BUTTON is on board, do not define additional LEDs/BUTTONs in BSP. This allows for unavailable LEDs/BUTTONs to be caught as a compile error.
+
+    // Do not do the following
```

Comment:
this should also provide what it should do first, and then not defining LED or BUTTONS as clones just to have all 1-4  available.

Or rather, the above section should contain a note that each of these should contain an unique pin (so no duplications allowed).

Reply:
@0xc0170 The current pin naming scheme across various Mbed enabled boards are quite vague and different across different boards. The purpose of this document is to get a starting point on describing the pin naming scheme for future boards (and possibly existing boards). With the differences in pin naming convention, there is a fake sense of portability across various Mbed enabled boards. This document writes down guidelines on how to better achieve that portability in a real sense when defining the pin names that are exposed to customers.

Reply:
Hm, I would just point out the "anti-paterns" or what not to do rather than give examples.

Reply:
@bulislaw I have collected lists of board support packages that define these pins differently for the various cases. Providing examples seemed like a more definite way of writing down guidelines.

Reply:
> this should also provide what it should do first, and then not defining LED or BUTTONS as clones just to have all 1-4 available.
> 
> Or rather, the above section should contain a note that each of these should contain an unique pin (so no duplications allowed).

https://github.com/ARMmbed/mbed-os/pull/12838/commits/63a3c8f59c24f96e2d2f2bf9dec17f4671f86571

Reply:
I think we all agree with that. We could write examples. It would be great if we could somehow catch these duplicates.